src/Chunk.ts

```diff
@@ -757,15 +754,15 @@ export default class Chunk {
 
 		// add in hashes of all dependent chunks and resolved external ids
 		this.postVisitChunkDependencies(dep => {
-			if (dep instanceof ExternalModule) hash.update(':' + dep.renderPath);
+			if (dep instanceof ExternalModule) hash.update(':' + dep.id);
```

Comment:
The issue here from a caching perspective is that now the hash is independent of the resolved external path. The worry with this is that changing the `output.paths` map or function implementation won't alter the hash. This is why this was carefully constructed this way with `renderPath`.

Because `output.paths` can be a function, we cannot compute a hash of the `output.paths` input to invalidate it that way either (even using `output.paths.toString()` is insufficient as it could have a closure on another function that changes.

Ideally we want to keep the renderPath stuff in pre-render. Is there an issue doing this? I will put some thought to it as well.